kernel/libterm/src/lib.rs

```diff
@@ -193,7 +193,7 @@ impl Terminal {
             if new_line_indices.len() == 0 {
                 // indicates that the text is just one continuous string with no newlines and will therefore fill the buffer completely
                 end_idx += buffer_height * buffer_width;
-                if end_idx <= self.scrollback_buffer.len() -1 {
+                if end_idx == self.scrollback_buffer.len() {
```

Comment:
this changes the logic, right?

Reply:
oh! I think it should have been < rather than == :facepalm: